From 9b8f1c9a1154601729ecab7e76ba7bf6952e2ba7 Mon Sep 17 00:00:00 2001
From: Limeng <Meng.Li@windriver.com>
Date: Tue, 19 Nov 2019 14:13:15 +0800
Subject: [PATCH 07/40] Revert "can: xilinx: support for remote frame
 transmission in can."

This reverts commit e3e1bf411638772c12c8dac76ae2a700488abf44.
---
 drivers/net/can/xilinx_can.c | 12 ++++--------
 1 file changed, 4 insertions(+), 8 deletions(-)

diff --git a/drivers/net/can/xilinx_can.c b/drivers/net/can/xilinx_can.c
index 11776ba650cd..d1ca52886397 100644
--- a/drivers/net/can/xilinx_can.c
+++ b/drivers/net/can/xilinx_can.c
@@ -669,14 +669,10 @@ static int xcan_start_xmit(struct sk_buff *skb, struct net_device *ndev)
 		if (buffnr == -1)
 			netif_stop_queue(ndev);
 	} else {
-		if (!(cf->can_id & CAN_RTR_FLAG)) {
-			if (cf->len > 0)
-				data[0] =
-				be32_to_cpup((__be32 *)(cf->data + 0));
-			if (cf->len > 4)
-				data[1] =
-				be32_to_cpup((__be32 *)(cf->data + 4));
-		}
+		if (cf->len > 0)
+			data[0] = be32_to_cpup((__be32 *)(cf->data + 0));
+		if (cf->len > 4)
+			data[1] = be32_to_cpup((__be32 *)(cf->data + 4));
 
 		can_put_echo_skb(skb, ndev, priv->tx_head % priv->tx_max);
 		priv->tx_head++;
-- 
2.17.1

